src/tasks.js

```diff
@@ -1,19 +1,33 @@
 const generateId = () => (Math.random() + 1).toString(36).substring(7);
 
-export const Task = (description) => {
-  const completedStatus = false;
-  const id = generateId();
+const generateIndex = () => {
+  if (localStorage.getItem('tasksArray') === null) {
+    return 0;
+  }
+  const storedTasksArray = JSON.parse(localStorage.getItem('tasksArray'));
+  return storedTasksArray.length;
+};
+
+export const Task = (taskData) => {
+  const {
+    description, completedStatus = false, id = generateId(), index = generateIndex(),
+  } = taskData;
   function changeCompletedStatus() {
     this.completedStatus = !this.completedStatus;
   }
+  function setDescription(newDescription) {
+    this.description = newDescription;
+  }
   return {
-    description, completedStatus, id, changeCompletedStatus,
+    description, completedStatus, id, changeCompletedStatus, setDescription, index,
   };
 };
 
-const taskOne = Task('Go groceries shopping');
-const taskTwo = Task('Wash the plates');
-const taskThree = Task('Do my laundry');
+const parseTasks = (tasks) => tasks.map(Task);
+
+// const taskOne = Task({ description: 'Go groceries shopping' });
+// const taskTwo = Task({ description: 'Wash the plates' });
+// const taskThree = Task({ description: 'Do my laundry' });
```

Comment:
- [ ] I think it would be better to remove the code instead of commenting from this and all other place as well. It usually makes your codebase ugly and messy so try avoiding pushing the commented code in production.